storage-manager/src/S3Storage.cpp

```diff
@@ -99,7 +102,7 @@ const char* s3err_msgs[] = {"All is well",
                             "Authentication failed, token has expired",
                             "Configured bucket does not match endpoint location"};
 
-S3Storage::ScopedConnection::ScopedConnection(S3Storage* s, ms3_st* m) : s3(s), conn(m)
+S3Storage::ScopedConnection::ScopedConnection(S3Storage* s, shared_ptr<S3Storage::Connection> m) : s3(s), conn(m)
```

Comment:
Why boost and not std::shared_ptr @mariadb-AlexeyAntipovsky ?

Reply:
It is `std::shared_ptr`